### PR TITLE
Fix ci artifacts not being uploaded when a job fails

### LINKED ISSUE #116: actions: test artefacts not being generated when tests fail

### PR DESCRIPTION
**Pull request by ttoino**
Closes #116.

# Review checklist
-   [ ] Properly documents API changes in `docs/openapi.yml`
-   [ ] Contains enough appropriate tests
-   [ ] Behavior is as expected
-   [ ] Clean, well structured code

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -42,6 +42,7 @@ jobs:
         with:
           name: lint-reports
           path: build/reports/ktlint
+        if: ${{ always() }}
```
Comment (limwa) [inline .github/workflows/ci.yml]: Expressions in an if-statement don't require the `${{ }}` syntax
```suggestion
        if: always()
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -75,3 +76,4 @@ jobs:
           path: |
             build/reports/tests/test
             build/reports/jacoco/test
+        if: ${{ always() }}
```
Comment (limwa) [inline .github/workflows/ci.yml]: Same here!
```suggestion
        if: always()
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (limwa) [reply .github/workflows/ci.yml]: I don't think this is that big of a deal, I thought I would just let you know!

**Review by limwa — APPROVED** [review]
LGTM 🚀

**Review by BrunoRosendo — APPROVED** [review]

**Pull request merged**